includes/class-wc-database-locks.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly
+}
+
+/**
+ * Allows the DB to be locked to avoid concurrent events doing the same thing.
+ *
+ * @since    3.2.0
+ * @package  WooCommerce/Classes
+ * @category Class
+ * @author   Automattic
+ */
+class WC_Database_Locks {
+
+	/**
+	 * Request ID.
+	 * @var string
+	 */
+	private static $request_id = '';
+
+	/**
+	 * Get ID of this request.
+	 *
+	 * @return string
+	 */
+	public static function get_request_id() {
+		if ( ! self::$request_id ) {
+			self::$request_id = uniqid( wp_rand(), true );
+			add_action( 'shutdown', __CLASS__, 'release_all_locks' );
```

Comment:
What is the purpose of tying a request id to these and releasing locks like this? Will we not have use cases where we would want a lock to possibly persist across requests? Is it because we don't trust people to call `release_lock` when they are done?